Sources/ApolloCodegenLib/IR/IR+RootFieldBuilder.swift

```diff
@@ -230,8 +230,29 @@ extension IR {
     }
 
     private func scopeCondition(
-      for conditionalSelectionSet: ConditionallyIncludable,
+      for inlineFragment: CompilationResult.InlineFragment,
       in parentTypePath: SelectionSet.TypeInfo
+    ) -> ScopeCondition? {
+      scopeCondition(for: inlineFragment, in: parentTypePath, isDeferred: inlineFragment.isDeferred)
+    }
+
+    private func scopeCondition(
+      for namedFragment: CompilationResult.FragmentSpread,
+      in parentTypePath: SelectionSet.TypeInfo
+    ) -> ScopeCondition? {
+      let matchedParentTypes = (parentTypePath.parentType == namedFragment.parentType)
+      let matchedInclusionConditions = (parentTypePath.inclusionConditions == InclusionConditions(namedFragment.inclusionConditions))
+
+      let scopedIsDeferred = matchedParentTypes && matchedInclusionConditions
+      ? namedFragment.isDeferred : false
+
+      return scopeCondition(for: namedFragment, in: parentTypePath, isDeferred: scopedIsDeferred)
+    }
+
+    private func scopeCondition(
+      for conditionalSelectionSet: ConditionallyIncludable,
+      in parentTypePath: SelectionSet.TypeInfo,
+      isDeferred: CompilationResult.IsDeferred
```

Comment:
We should change the type of this parameter to `IR.IsDeferred` and do ``IsDeferred(compilationResult: namedFragment.isDeferred)` on line 247.